File: app/src/main/java/com/yang/connetwork/GetDataListApi.java
package com.yang.connetwork;

import com.yang.connetwork.network.NetWorkApi;
import com.yang.connetwork.network.NetWorkBasicResponse;

import org.jetbrains.annotations.NotNull;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author yangzc
 * @data 2020/1/19 14:22
 * @desc
 */
public class GetDataListApi extends NetWorkApi {
    private static final String mURL = "2/json";


    public GetDataListApi() {
        super(mURL);
    }

    @Override
    public int getHttpRequestType() {
        return NetWorkApi.Companion.getHTTP_REQUEST_TYPE_GET();
    }


    @Override
    public NetWorkBasicResponse parseResponseBase(JSONObject json) {
        try {
            return new GetDataListResponse(json);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return null;
    }

    public class GetDataListResponse extends NetWorkBasicResponse {

        public GetDataListResponse(@NotNull JSONObject json) throws JSONException {
            super(json);
        }
    }


}

File: app/src/main/java/com/yang/connetwork/GetDataListApi.kt
package com.yang.connetwork

import com.yang.connetwork.network.NetWorkApi
import com.yang.connetwork.network.NetWorkBasicResponse
import org.json.JSONException
import org.json.JSONObject


/**
 * @author yangzc
 * @data 2020/1/19 14:22
 * @desc
 */
class GetDataListApi : NetWorkApi(mURL) {

    override val httpRequestType: Int
        get() = HTTP_REQUEST_TYPE_GET


    override fun parseResponseBase(json: JSONObject): NetWorkBasicResponse? {
        try {
            return GetDataListResponse(json)
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return null
    }

    inner class GetDataListResponse @Throws(JSONException::class)
    constructor(json: JSONObject) : NetWorkBasicResponse(json) {
        var data: String = json.getJSONObject("data").getJSONArray("datas").toString()

    }

    companion object {
        private val mURL = "2/json"
    }


}
